### DISCUSSION TITLE
@njit doesn't respond to dpctl.device_context

### DISCUSSION THREAD

**Discussion by samaid** [IntelPython/numba-dppy#613] [Ideas]
I suggest to change import numba behavior to enable numba_dppy pipeline if import dpctl and/or import numba_dppy is present.

This is how we enabled SDC pipeline. If the following is present in Python script

import pandas

then 
import numba

triggers SDC compilation pipeline.

We can do the same for Numba-DPPY.

If
import dpctl
Or
import numba_dppy
is present then
import numba

Should trigger Numba-DPPY compilation pipeline.

This is the only reasonable way that I see @njit to respect dpctl device/queue management.

**Comment by PokhodenkoSA** [comment]
Please, note that Numba does not allow change `compilation pipeline` implicitly. But it allows fill typing system with new types and SDC uses it to fill Numba types with pandas types.
What I could propose is to use import substitution like in modin, dpnp and others - change `import numba` to `from numba-dppy import numba` and use `numba.jit` provided by numba-dppy.

**Reply by PokhodenkoSA** [reply]
As I understand it worth to try the approach w/ `from numba_dppy import numba`. See https://github.com/IntelPython/numba-dppy/issues/620.

**Comment by diptorupd** [comment]
@samaid  The device_context is deprecated and will be removed. 

In 0.20 we released a new decorator `@dpjit` that is the intended way to use numba-dpex's dpnp compilation and compute-follows-data functionalities.
